miden/src/cli/run.rs

```diff
@@ -171,7 +171,7 @@ fn run_masm_program(params: &RunCmd) -> Result<(ExecutionTrace, [u8; 32]), Repor
         Some(params.max_cycles),
         params.expected_cycles,
         params.trace,
-        params.debug,
+        !params.release,
```

Comment:
`get_masm_program()` above should also have a flag `debug_on`, as it currently always uses `ProgramFile::compile(Debug::On)` internally. It should instead set `Debug::On/Off` based on the `debug_on` flag